pyne/alara.py

```diff
@@ -890,5 +893,365 @@ def _find_phsrc_dc(idc, phtn_src_dc):
+    phtn_src_file = os.path.join(run_dir, "phtn_src")
+    _gt_write_inp(run_dir, data_dir, mats, num_n_groups, flux_magnitudes, 
+                  irr_times, decay_times, input_file, matlib_file,
+                  fluxin_file, phtn_src_file, num_p_groups, p_bins)
+
+    # Run ALARA
+    sub = subprocess.check_output(['alara', input_file], stderr=subprocess.STDOUT)
+    return phtn_src_file
+
+def calc_eta(data_dir, mats, neutron_spectrum, flux_magnitudes, irr_times,
+             decay_times, num_p_groups, p_bins, run_dir, clean):
+    """
+    Function that returns eta values (SNILB check result) for each material 
+    and each decay time
+    
+    Parameters
+    ----------
+    data_dir : str
+        Path to directory containing nuclib and fendl files
+    mats : list
+        List of PyNE material objects of materials in the geometry
+    neutron_spectrum : numpy array
+        Neutron energy spectrum (length is equal to number of n energy groups)
+    flux_magnitudes : list
+        Neutron flux normalization
+    irr_times : list
+        Irradiation times [s]
+    decay_times : list
+        Decay times [s]
+    num_p_groups : int
+        The number of photon energy groups for ALARA calculation
+    p_bins: numpy array
+       Photon energy bin bounds
+    run_dir: str
+        Path to write ALARA input and output files    
+    clean : bool
+        If True, remove run_dir
+        
+    Returns
+    ----------
+    eta : numpy array
+        eta value per photon group for each material listed.  
+        This is a 3D array [mat, decay_time, num_p_groups + 1]
+    """
+    num_n_groups = len(neutron_spectrum)
+    num_mats = len(mats)
+    num_decay_times = len(decay_times)
+    
+    # Run ALARA
+    if not os.path.exists(run_dir):
+        os.makedirs(run_dir)    
+    phtn_src_file = _gt_alara(data_dir, mats, neutron_spectrum, flux_magnitudes, 
+                              irr_times, decay_times, num_p_groups, p_bins, run_dir)
+    # Parse ALARA output
+    sup = np.zeros(shape=(num_mats, num_decay_times, num_p_groups + 1))
+    tot = np.zeros(shape=(num_mats, num_decay_times, num_p_groups + 1))
+    zero = np.zeros(shape=(num_mats, num_decay_times, num_p_groups + 1))
+    with open(phtn_src_file, 'r') as f:
+        # Initiate a block number
+        i = 0
+        for line in f.readlines():
+            l = line.split()
+            if l[0] == "TOTAL" and l[1] != "shutdown":
+                row = np.array([float(x) for x in l[3:]])
+                row_sum = np.sum(row)
+                # Material index will change every (num_n_groups + 2) * num_decay_times
+                entries_per_material = (num_n_groups + 2) * num_decay_times
+                m = int(np.floor(float(i) / entries_per_material))
+                dt = i % num_decay_times
+                n = int(np.floor(i/float(num_decay_times))) % (num_n_groups + 2)
+                if n == num_n_groups:
+                    tot[m, dt, :] = np.hstack([row, row_sum])
```

Comment:
isn't `row_sum` a scalar?  Can't you just append it?

Reply:
Also - it appears that you always append it, and the only difference is where to assign it. - move the append outside the conditional